packages/kit/test/apps/basics/test/client.test.js

```diff
@@ -619,6 +619,36 @@ test.describe('Invalidation', () => {
 });
 
 test.describe('data-sveltekit attributes', () => {
+	test('data-sveltekit-preload-code', async ({ page }) => {
+		/** @type {string[]} */
+		const requests = [];
+		page.on('request', (r) => {
+			requests.push(r.url());
+		});
+
+		// eager
+		await page.goto('/data-sveltekit/preload-code');
+		expect(requests.length).toBeGreaterThanOrEqual(1);
+
+		// viewport
+		requests.length = 0;
+		page.locator('#viewport').scrollIntoViewIfNeeded();
+		await Promise.all([page.waitForTimeout(100), page.waitForLoadState('networkidle')]);
```

Comment:
`waitForTimeout` is an anti-pattern. can we avoid it? why do we need both it and `networkidle`?

Reply:
> `waitForTimeout` is an anti-pattern. can we avoid it? why do we need both it and `networkidle`?

It was used in the `data-sveltekit-preload-data` tests because the hover preload waits before preloading a link. I just copied the test conditions for consistency.

 I'll remove the waitForTimeout because they are not needed here

Reply:
I think it may be needed here just so that there's time for the network request to fire. Removing them causes the tests to fail with 0 network requests.